### ISSUE TITLE
Mono: multiple problems with exported variables of Texture type 

### ISSUE THREAD

**Issue by tagcup** [godotengine/godot#12043]
(using current master)

* Exported Texture variables can't be set through the property editor: clicking on the dropdown menu (to set an image) does nothing.
* Cause a failed assertion at the end

```
ERROR: ~List: Condition ' _first != __null ' is true.
   At: core/self_list.h:99.
ERROR: getornull: Condition ' !id_map.has(p_rid.get_data()) ' is true. returned: __null
   At: core/rid.h:164.
ERROR: instance_remove_dependency: Condition ' !inst ' is true.
   At: drivers/gles3/rasterizer_storage_gles3.cpp:5825.
```

Example project in #11955 demonstrates these issues

**Comment by tagcup** [comment]
The first one is actually not specific to Texture, any type of mono-exported variable can't be edited in the property editor.

**Comment by neikeq** [comment]
> * Exported Texture variables can't be set through the property editor: clicking on the dropdown menu (to set an image) does nothing.

This one is fixed by #12178
